nbconvert/preprocessors/numbered_headings.py

```diff
@@ -0,0 +1,63 @@
+"""
+Preprocessor that transforms markdown cells: Insert numbering in from of heading
+"""
+
+from traitlets.log import get_logger
+
+from nbconvert.preprocessors.base import Preprocessor
+
+logger = get_logger()
+
+try:  # for Mistune >= 3.0
+    import mistune
+    from mistune.core import BlockState
+    from mistune.renderers.markdown import MarkdownRenderer
+except ImportError:  # for Mistune >= 2.0
+    logger.error("NumberedHeadingsPreprocessor requires mistune >= 3")
```

Comment:
I would set a variable here (and in the try block too), like:

```suggestion
    mistune_3_available = False
```

and only emit the error (throw?) if someone tries to initialize `NumberedHeadingsPreprocessor` (this is in `__init__`). This is because logging on import of preprocessors would affect these who do not wish to use `NumberedHeadingsPreprocessor`

Reply:
Went for throwing an exception ( for some reason log output does not seem displayed)